### ISSUE TITLE
async Dictionary vs Array conflicts 

### ISSUE THREAD

**Issue by ORESoftware** [DefinitelyTyped/DefinitelyTyped#24543]
@jeremyhon @sasanj @fenying @kasuparu @rlindgren 

- [x] I tried using the `@types/async` package and had problems.

Looking at the typings, I see this:

```ts
export function eachLimit<T, E>(arr: T[] | IterableIterator<T>, limit: number, iterator: AsyncIterator<T, E>, callback?: ErrorCallback<E>): void;

export function eachLimit<T, E>(arr: Dictionary<T>, limit: number, iterator: AsyncIterator<T, E>, callback?: ErrorCallback<E>): void;

```
<br>

it appears as though these aren't working together. I have this problem:

<kbd>
<img width="837" alt="screenshot 2018-03-26 11 59 56" src="https://user-images.githubusercontent.com/11139560/37926818-38e3de02-30ed-11e8-86fc-adb361e920b1.png">
</kbd>

<br>
<br>

the two `eachLimit` definitions have the same signature except for the first argument. My IDE (Webstorm) complains because it looks like my code fails to meet the second definition, although it should be meeting the first definition of `eachLimit`.

_**I know this because if I comment out the second definition, my problem goes away:**_

<img width="1086" alt="screenshot 2018-03-26 12 01 36" src="https://user-images.githubusercontent.com/11139560/37926924-7bda03b2-30ed-11e8-8559-6b5febb19fe0.png">

Here is the definition of AC:

```ts
import {Collection} from 'mongoose';
export type AC =  Array<Collection>;
```

**How do we solve this problem?**

**Comment by jeremyhon** [comment]
Hi, I actually faced this problem and have already fixed it in a recent PR: #24188 

This fix should be in the most recent version of `@types/async: 2.0.48`, are you using that version?

**Comment by orta** [comment]
Hi thread, we're moving DefinitelyTyped to use [GitHub Discussions](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/53377) for conversations the `@types` modules in DefinitelyTyped.

To help with the transition, we're closing all issues which haven't had activity in the last 6 months, which includes this issue. If you think closing this issue is a mistake, please pop into the [TypeScript Community Discord](https://discord.gg/typescript) and mention the issue in the `definitely-typed` channel.
